common/storage/persistable.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __COMMON_STORAGE_PERSISTABLE_H__
+#define __COMMON_STORAGE_PERSISTABLE_H__
+
+#include <stdint.h>
+#include "common/trinary/trit_array.h"
+#include "common/errors.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct persistable_base persistable_t;
+
+typedef struct {
```

Comment:
not fully following here. could you explain your approach?

Reply:
Every view model should extend `persistable_t`
the base struct holds a ptr to the opaque ptr of the derived struct
so when you call `stor_store/stor_load` you can get the key/value of the underlying object

Reply:
Also, i'm not exactly sure about `table_name`, but some table identifier needs to be included

Reply:
How about we make this not so generic? 
It's perfectly fine to limit the available 'tables' via a enum in code and then just have a single method for storing/loading everything imho.
The `vtable` adds significant overhead.

Reply:
Alright, that might be simpler